webapp/platform/components/src/generic_modal/generic_modal.tsx

```diff
@@ -67,249 +68,242 @@ export type Props = {
-        if (this.props.handleCancel) {
-            this.props.handleCancel();
-        }
-    };
+        props.handleCancel?.();
+    }, [autoCloseOnCancelButton, onHide, props.handleCancel]);
 
-    handleConfirm = (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
+    const handleConfirm = useCallback((event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
         event.preventDefault();
-        if (this.props.autoCloseOnConfirmButton) {
-            this.onHide();
-        }
-        if (this.props.handleConfirm) {
-            this.props.handleConfirm();
+        if (autoCloseOnConfirmButton) {
+            onHide();
         }
-    };
+        props.handleConfirm?.();
+    }, [autoCloseOnConfirmButton, onHide, props.handleConfirm]);
 
-    private onEnterKeyDown = (event: React.KeyboardEvent<HTMLDivElement>) => {
+    const onEnterKeyDown = useCallback((event: React.KeyboardEvent<HTMLDivElement>) => {
         if (event.key === 'Enter') {
             if (event.nativeEvent.isComposing) {
                 return;
             }
-            if (this.props.handleConfirm && this.props.autoCloseOnConfirmButton) {
-                this.onHide();
-            }
-            if (this.props.handleEnterKeyPress) {
-                this.props.handleEnterKeyPress();
+            if (props.handleConfirm && autoCloseOnConfirmButton) {
+                onHide();
             }
+            props.handleEnterKeyPress?.();
         }
-        this.props.handleKeydown?.(event);
-    };
+        props.handleKeydown?.(event);
+    }, [props.handleConfirm, autoCloseOnConfirmButton, onHide, props.handleEnterKeyPress, props.handleKeydown]);
 
-    render() {
-        let confirmButton;
-        if (this.props.handleConfirm) {
-            const isConfirmOrDeleteClassName = this.props.isDeleteModal ? 'delete' : 'confirm';
-            let confirmButtonText: React.ReactNode = (
-                <FormattedMessage
-                    id='generic_modal.confirm'
-                    defaultMessage='Confirm'
-                />
-            );
-            if (this.props.confirmButtonText) {
-                confirmButtonText = this.props.confirmButtonText;
-            }
-
-            confirmButton = (
-                <button
-                    autoFocus={this.props.autoFocusConfirmButton}
-                    type='submit'
-                    className={classNames('GenericModal__button btn btn-primary', isConfirmOrDeleteClassName, this.props.confirmButtonClassName, {
-                        disabled: this.props.isConfirmDisabled,
-                    })}
-                    onClick={this.handleConfirm}
-                    disabled={this.props.isConfirmDisabled}
-                >
-                    {confirmButtonText}
-                </button>
-            );
+    // Build confirm button if provided.
+    let confirmButton;
+    if (props.handleConfirm) {
+        const buttonTypeClass = props.isDeleteModal ? 'delete' : 'confirm';
```

Comment:
One of the good things about destructuring the props in the arguments of the function is that you don't do this where in some places you use the destructured version, and in some places the `props.` version.